go-controller/pkg/ovn/controller/network_qos/network_qos.go

```diff
@@ -29,50 +29,38 @@ import (
 func (c *Controller) processNextNQOSWorkItem(wg *sync.WaitGroup) bool {
 	wg.Add(1)
 	defer wg.Done()
-	nqosKey, quit := c.nqosQueue.Get()
+	nqos, quit := c.nqosQueue.Get()
```

Comment:
we want to keep the queues using keys, and then in here once you lock the key, you will list the nqos from the informer cache and try to sync it. Otherwise by queuing nqos events from other handlers, you can essentially modify the real order of events as those handlers are asynchronous, and we can end up with incorrect configuration. Take a look at:

https://github.com/ovn-kubernetes/ovn-kubernetes/blob/master/go-controller/pkg/ovn/controller/apbroute/external_controller_policy.go#L40

Reply:
ack, we've reverted to use string key.